core/src/tag_utils.rs

```diff
@@ -151,11 +151,59 @@ impl SwfMovie {
         Self::from_data(&data, url.into(), loader_url)
     }
 
-    /// Construct a movie based on the contents of the SWF datastream.
+    /// Construct a movie based on the contents of a datastream.
     pub fn from_data(
         swf_data: &[u8],
         url: String,
         loader_url: Option<String>,
+    ) -> Result<Self, Error> {
+        match Self::filter_from_projector_bundle(swf_data) {
+            Some(extracted_swf) => Self::from_swf_data(&extracted_swf, url, loader_url),
+            None => Self::from_swf_data(swf_data, url, loader_url),
+        }
+    }
+
+    /// Get an SWF from data if the data is a swf/projector bundle
+    fn filter_from_projector_bundle(data: &[u8]) -> Option<Vec<u8>> {
```

Comment:
Not important, just a little note: Do you think it would be possible to make this return an `Option<&[u8]>` instead of an `Option<Vec<u8>>`? Just to avoid a copy. Doing this might involve a lifetime annotation or two. I don't have a definite solution in my head, maybe it's not feasible.
It's also fine if it stays as is though.

Reply:
You know... I think it is (haven't had time to test fully, but a little playing around seemed promising).

I probably spent a week trying to figure out how to return a slice but kept failing. Adding the Option/Some seems to make it happy, I'll have to look more into that (pretty new to Rust).

Reply:
You can't return a slice itself without boxing, but you can return the borrow to it.